components/widgets/menu/menu.jsx

```diff
@@ -32,11 +32,11 @@ export default class Menu extends React.PureComponent {
     render() {
         const {children, openUp, openLeft, onMobile, id, ariaLabel} = this.props;
         const styles = {};
-        if (openLeft && (!isMobile() || !onMobile)) {
+        if (openLeft && (!isMobile() || onMobile === false)) {
```

Comment:
That fixes the immediate problem, but I was curious why this might be undefined and so I did a quick search to see where this prop might be getting passed in. I couldn't find it anywhere. I did find a prop called `mobile` getting passed into menu.jsx in `main_menu.jsx`. Is there perhaps a typo here? If not do you know what the purpose of the `onMobile` prop is?

Reply:
I added it to have a condition of whether we would like the menu to open up as a dropdown on mobile. and not take up the whole width/height.
Though the whole code is not in there, so we can remove the onMobile thing if we want, and add it back later.

Reply:
Okay, so it's an override to allow for changing the default behaviour at a later date?

@jespino, looks like you might have had a similar idea in `main_menu.jsx` on line 142? If so, should we consolidate these two implementations?

Either way, I'm wondering if we should maybe change `onMobile` to `mobile` as at a glance, `onMobile` reads a bit like an action like `onClick`?

Reply:
Yup, we can do that.

Reply:
Looks like a typo and the correct props is `mobile`, although I'm at lost what's the difference with the `isMobile()` utlity. And it's being hard-coded only at https://github.com/asaadmahmood/mattermost-webapp/blob/MM-14787/components/sidebar_right_menu/sidebar_right_menu.jsx#L78

Reply:
It's something I added for later, it's not really anything for this PR.
Mobile is just a manual way of checking if we want the dropdown to open on mobile. We can add the whole implementation later.